src/WebJobs.Script/BindingExtensions/ExtensionsManager.cs

```diff
@@ -0,0 +1,249 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Description.DotNet;
+using Microsoft.Azure.WebJobs.Script.Models;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Extensions.Logging;
+using static Microsoft.Azure.WebJobs.Script.ScriptConstants;
+
+namespace Microsoft.Azure.WebJobs.Script.BindingExtensions
+{
+    public class ExtensionsManager : IExtensionsManager
+    {
+        private readonly string _scriptRootPath;
+        private readonly TraceWriter _traceWriter;
+        private readonly ILogger _logger;
+
+        public ExtensionsManager(string scriptRootPath, TraceWriter traceWriter, ILogger logger)
+        {
+            _scriptRootPath = scriptRootPath;
+            _traceWriter = traceWriter;
+            _logger = logger;
+        }
+
+        internal string ProjectPath => Path.Combine(_scriptRootPath, ExtensionsProjectFileName);
+
+        public async Task AddExtensions(params ExtensionPackageReference[] references)
+        {
+            if (!references.Any())
+            {
+                return;
+            }
+
+            await Task.Run(async () =>
```

Comment:
>Task.Run [](start = 18, length = 8)

Why the extra Task.Run?

Reply:
This is really not extra. This starts a Task that runs some synchronous IO bound operations (creating project file, etc.)

Moving this to the IO bound methods to simplify this.